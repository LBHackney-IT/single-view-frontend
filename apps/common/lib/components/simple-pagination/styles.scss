@import "~lbh-frontend/lbh/base";
@import "~govuk-frontend/govuk/helpers/media-queries";

.lbh-simple-pagination {
  display: flex;
  flex-direction: row;
  border-top: 1px solid lbh-colour("lbh-grey-3");
  position: relative;

  * + * {
    margin-top: 0;
  }

  &__link {
    background: lbh-colour("lbh-white");
    border: none;
    text-decoration: none;
    text-align: left;
    display: inline-block;
    flex: 1;
    padding: 10px 0px 10px 30px;
    @include lbh-body-m;
    font-weight: $lbh-font-weight-bold;
    cursor: pointer;
    color: lbh-colour("lbh-link");

    &:hover {
      color: lbh-colour("lbh-link-hover");
      text-decoration: none;

      .lbh-simple-pagination__title {
        text-decoration: underline;
      }
    }

    &:active {
      color: lbh-colour("lbh-link-active");
    }

    &:focus {
      outline: none;
      background: lbh-colour("lbh-primary-focus");
      color: lbh-colour("lbh-text");
      box-shadow: 0 -2px lbh-colour("lbh-primary-focus"), 0 4px lbh-colour("lbh-text");

      svg {
        stroke: lbh-colour("lbh-text");
      }
    }

    svg {
      position: absolute;
      left: 5px;
      top: 17px;
      stroke: lbh-colour("lbh-grey-3");
    }

    &--next {
      padding: 10px 30px 10px 0px;
      text-align: right;

      svg {
        position: absolute;
        left: inherit;
        right: 5px;
      }
    }
  }

  &__title {
    display: block;

    @include lbh-font;
    @include lbh-rem(font-size, 13);
    font-weight: $lbh-font-weight-regular;
    @include lbh-rem(line-height, 19);

    @include govuk-media-query($from: tablet) {
      @include lbh-rem(font-size, 16);
      @include lbh-rem(line-height, 24);
    }
  }
}
