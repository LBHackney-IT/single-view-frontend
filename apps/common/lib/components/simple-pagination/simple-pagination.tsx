import React, { ComponentPropsWithoutRef, forwardRef } from "react";

import cn from "classnames";

import type * as Polymorphic from "@radix-ui/react-polymorphic";
import "./styles.scss";

export const SimplePagination = forwardRef<HTMLElement, ComponentPropsWithoutRef<"nav">>(
  function SimplePagination({ className, ...props }, ref) {
    return (
      <nav ref={ref} className={cn("lbh-simple-pagination", className)} {...props} />
    );
  },
);

export interface SimplePaginationButtonProps {
  title?: string;
  variant: "previous" | "next";
}

export type SimplePaginationButtonComponent = Polymorphic.ForwardRefComponent<
  "a",
  SimplePaginationButtonProps
>;

export const SimplePaginationButton: SimplePaginationButtonComponent = forwardRef(
  function SimplePaginationButton(
    { as: SimplePaginationComp = "a", variant, className, title, children, ...props },
    ref,
  ) {
    return (
      <SimplePaginationComp
        ref={ref}
        className={cn(
          "lbh-simple-pagination__link",
          { "lbh-simple-pagination__link--next": variant === "next" },
          className,
        )}
        rel={variant}
        {...props}
      >
        {variant === "previous" ? (
          <svg width="11" height="19" viewBox="0 0 11 19" fill="none">
            <path d="M10 1L2 9.5L10 18" strokeWidth="2" />
          </svg>
        ) : null}
        {children}
        {title ? <span className="lbh-simple-pagination__title">{title}</span> : null}
        {variant === "next" ? (
          <svg width="11" height="19" viewBox="0 0 11 19" fill="none">
            <path d="M1 18L9 9.5L1 1" strokeWidth="2" />
          </svg>
        ) : null}
      </SimplePaginationComp>
    );
  },
);
