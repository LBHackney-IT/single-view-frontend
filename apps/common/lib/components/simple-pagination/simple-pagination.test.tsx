import React from "react";

import { render, testA11y } from "@hackney/mtfh-test-utils";

import { SimplePagination, SimplePaginationButton } from "./simple-pagination";

test("it renders the nav with buttons", async () => {
  const { container } = render(
    <SimplePagination>
      <SimplePaginationButton href="/prev" variant="previous">
        Prev
      </SimplePaginationButton>
      <SimplePaginationButton href="/next" variant="next">
        Next
      </SimplePaginationButton>
    </SimplePagination>,
  );
  expect(container).toMatchSnapshot();
  await testA11y(container);
});

test("it renders the nav with buttons and titles", () => {
  const { container } = render(
    <SimplePagination>
      <SimplePaginationButton href="/prev" variant="previous" title="1 of 3">
        Prev
      </SimplePaginationButton>
      <SimplePaginationButton href="/next" variant="next" title="3 of 3">
        Next
      </SimplePaginationButton>
    </SimplePagination>,
  );
  expect(container).toMatchSnapshot();
});
