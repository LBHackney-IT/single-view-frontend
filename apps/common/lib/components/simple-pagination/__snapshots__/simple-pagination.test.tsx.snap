// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders the nav with buttons 1`] = `
<div>
  <nav
    class="lbh-simple-pagination"
  >
    <a
      class="lbh-simple-pagination__link"
      href="/prev"
      rel="previous"
    >
      <svg
        fill="none"
        height="19"
        viewBox="0 0 11 19"
        width="11"
      >
        <path
          d="M10 1L2 9.5L10 18"
          stroke-width="2"
        />
      </svg>
      Prev
    </a>
    <a
      class="lbh-simple-pagination__link lbh-simple-pagination__link--next"
      href="/next"
      rel="next"
    >
      Next
      <svg
        fill="none"
        height="19"
        viewBox="0 0 11 19"
        width="11"
      >
        <path
          d="M1 18L9 9.5L1 1"
          stroke-width="2"
        />
      </svg>
    </a>
  </nav>
</div>
`;

exports[`it renders the nav with buttons and titles 1`] = `
<div>
  <nav
    class="lbh-simple-pagination"
  >
    <a
      class="lbh-simple-pagination__link"
      href="/prev"
      rel="previous"
    >
      <svg
        fill="none"
        height="19"
        viewBox="0 0 11 19"
        width="11"
      >
        <path
          d="M10 1L2 9.5L10 18"
          stroke-width="2"
        />
      </svg>
      Prev
      <span
        class="lbh-simple-pagination__title"
      >
        1 of 3
      </span>
    </a>
    <a
      class="lbh-simple-pagination__link lbh-simple-pagination__link--next"
      href="/next"
      rel="next"
    >
      Next
      <span
        class="lbh-simple-pagination__title"
      >
        3 of 3
      </span>
      <svg
        fill="none"
        height="19"
        viewBox="0 0 11 19"
        width="11"
      >
        <path
          d="M1 18L9 9.5L1 1"
          stroke-width="2"
        />
      </svg>
    </a>
  </nav>
</div>
`;
